### PR TITLE
fix: cookie being appended with comma

### PR DESCRIPTION
**Pull request by jayahariv**
* this should ideally add new cookies with semi-colons.
* https://developer.apple.com/documentation/foundation/nsmutableurlrequest/1408793-setvalue

### CODE REVIEW

File: Objective-C/Internal/Replicator/CBLWebSocket.mm [hunk 1]
```diff
@@ -376,12 +376,12 @@ - (void) _sendWebSocketRequest {
...
     slice sessionCookie = _options[kC4ReplicatorOptionCookies].asString();
     if (sessionCookie.buf)
-        [_logic addValue: sessionCookie.asNSString()  forHTTPHeaderField: @"Cookie"];
+        [_logic setValue: sessionCookie.asNSString()  forHTTPHeaderField: @"Cookie"];
     
     
     NSString* cookie = [_db getCookies: _remoteURL];
     if (cookie.length > 0)
-        [_logic addValue: cookie  forHTTPHeaderField: @"Cookie"];
+        [_logic setValue: cookie  forHTTPHeaderField: @"Cookie"];
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLWebSocket.mm:384]: How does this change work with multiple Cookies (one from kC4ReplicatorOptionCookies and the other from the cookie store)?

Comment (jayahariv) [reply Objective-C/Internal/Replicator/CBLWebSocket.mm:384]: it should ideally comes like cookieOne=value; cookieTwo=value2

Comment (pasin) [reply Objective-C/Internal/Replicator/CBLWebSocket.mm:384]: From the doc that you have linked:
```
The new value for the header field. Any existing value for the field is replaced by the new value.
```
Do you mean Cookie key is a special key that setting multiple values will be combined with semi-colons?

Comment (jayahariv) [reply Objective-C/Internal/Replicator/CBLWebSocket.mm:384]: if same cookie-key is used, it will override the previous set cookie.

Comment (jayahariv) [reply Objective-C/Internal/Replicator/CBLWebSocket.mm:384]: I get what you mean

Comment (pasin) [reply Objective-C/Internal/Replicator/CBLWebSocket.mm:384]: OK

**Review by pasin — APPROVED** [review]

**Pull request merged**